src/extension.ts

```diff
@@ -18,59 +18,38 @@ export async function activate(context: ExtensionContext) {
+    context.subscriptions.push(commands.registerCommand(CommandNames.RefreshPollingStatus, () => _extensionManager.RunCommand(() => _extensionManager.Team.RefreshPollingStatus())));
     context.subscriptions.push(commands.registerCommand(CommandNames.Reinitialize, () => _extensionManager.Reinitialize()));
 
     // TFVC Commands
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.UndoAll, () => {
-            _extensionManager.Tfvc.UndoAll();
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Undo, (...args) => {
-            _extensionManager.Tfvc.Undo(args);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Exclude, (...args) => {
-            _extensionManager.Tfvc.Exclude(args);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Include, (...args) => {
-            _extensionManager.Tfvc.Include(args);
-        }));
-    //Open is meant only to be used by the Tfvc source control provider when a user selects an item in the viewlet (restrict to one file)
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Open, (...args) => {
-            _extensionManager.Tfvc.Open(args ? args[0] : undefined);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.OpenDiff, (...args) => {
-            _extensionManager.Tfvc.OpenDiff(args ? args[0] : undefined);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.OpenFile, (...args) => {
-            _extensionManager.Tfvc.OpenFile(args ? args[0] : undefined);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Refresh, () => _extensionManager.Tfvc.Refresh()));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Rename, (...args) => {
-            _extensionManager.Tfvc.Rename(args ? args[0] : undefined);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.ShowOutput, () => _extensionManager.Tfvc.ShowOutput()));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.ResolveKeepYours, (...args) => {
-            _extensionManager.Tfvc.Resolve(args ? args[0] : undefined, AutoResolveType.KeepYours);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.ResolveTakeTheirs, (...args) => {
-            _extensionManager.Tfvc.Resolve(args ? args[0] : undefined, AutoResolveType.TakeTheirs);
-        }));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Checkin, () => _extensionManager.Tfvc.Checkin()));
-    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.Sync, () => _extensionManager.Tfvc.Sync()));
+    context.subscriptions.push(commands.registerCommand(TfvcCommandNames.UndoAll, () => _extensionManager.RunAsyncCommand(() => _extensionManager.Tfvc.UndoAll())));
```

Comment:
As it turns out, in this instance, I don't even need RunAsyncCommand. Another commit coming.